File: 14-longest-common-prefix/longest-common-prefix.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        if (strs == null || strs.Length == 0)
            return "";

        string result = "";

        for (int i = 0; i < strs[0].Length; i++) {
            char currentChar = strs[0][i];

            for (int j = 1; j < strs.Length; j++) {
                if (i >= strs[j].Length || strs[j][i] != currentChar) {
                    return result;
                }
            }

            result += currentChar;
        }

        return result;
    }
}

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        result = ""

        # We are iterrating over every index of the array strs
        for i in (range(len(strs[0]))):
            for string in strs:
                if i == len(string) or string[i] != strs[0][i]:
                    return result
            result += strs[0][i]
        return result
